WebUI/Data/DataAccess/ReviewService.cs

```diff
@@ -0,0 +1,65 @@
+﻿using ConestogaInsidersClub.Data.Models;
+using Microsoft.EntityFrameworkCore;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace ConestogaInsidersClub.Data.DataAccess
+{
+    public class ReviewService : IReviewService
+    {
+        private readonly ApplicationDbContext context;
+
+        public ReviewService(ApplicationDbContext context)
+        {
+            this.context = context;
+        }
+        public async Task AddReview(Review review)
+        {
+            await context.Reviews.AddAsync(review);
+            await context.SaveChangesAsync();
+        }
+
+        public async Task ApproveReview(Review review)
+        {
+            review.Approved = true;
+            context.Reviews.Update(review);
+            await context.SaveChangesAsync();
+        }
+
+        public async Task DeleteReview(Review review)
+        {
+            context.Reviews.Remove(review);
+            await context.SaveChangesAsync();
+        }
+
+        public async Task<List<Review>> GetApprovedGameReviews(int gameId)
+        {
+            return await context.Reviews.Where(a => a.GameId == gameId && a.Approved == true).ToListAsync();
+        }
+
+        public async Task<double> GetAverageRating(int gameId)
+        {
+            List<Review> reviews = await context.Reviews.Where(a => a.GameId == gameId && a.Approved == true).ToListAsync();
+            return reviews.Average(a => a.Rating);
+        }
+
+        public async Task<List<Review>> GetReviews()
```

Comment:
I would update the name of the method to make it clear that it returns "disapproved" reviews. Not sure if `GetDisapprovedReviews` is the best option, but I can't think of anything better 😓

Reply:
I agree, given you already have a `GetApprovedGameReviews` method.